includes/views/listing_contact.php

```diff
@@ -55,6 +57,10 @@ private function validate() {
             $this->errors[] = _x( 'Please enter a valid email.', 'contact-message', 'business-directory-plugin' );
         }
 
+        if ( ! $this->phone ) {
+            $this->errors[] = _x( 'Please enter your phone number.', 'contact-message', 'business-directory-plugin' );
```

Comment:
Let's keep the phone field optional. Maybe we can include a hook for people to add custom validation instead?

Reply:
@stephywells I have pushed an update that includes a filter. Should the validation message also be custom or should we leave it to the filter `wpbdp_contact_form_validation_errors` to handle this?

Reply:
I was thinking we could add a filter on the $this->errors array so anything could be added. We can add a code example on our site to make the phone required.

Reply:
@stephywells there is documentation of this here https://businessdirectoryplugin.com/knowledge-base/add-phone-number-to-listing-contact-form/ on how to add custom validation. Though I am not sure how users will take to having to use a filter to add validation messages for a file that comes with the plugin

Reply:
Oh, right. There's already a hook for this.

I think generally, best practice is to not require phone fields for things like this. The example here would cover this great:
```
function example_validate_custom_contact_field( $errors ) {
  if ( empty( $_POST['phone-number'] ) ) {
    $errors['phone-number'] = 'Phone number is required!';
  }
  return $errors;
}
add_filter( 'wpbdp_contact_form_validation_errors', 'example_validate_custom_contact_field' );
```

Ideally at some future point, we'll have an easier way to replace this with a Formidable form of something, so it can be completely customized.